iOS_SDK/OneSignalSDK/build_fat_framework.sh

```diff
@@ -1,52 +1,45 @@
-# Generates a universal/fat framework that can be used in multiple architectures (x86_64 and arm64) to support both simulator and actual devices.
-# Note: When complete, this build script takes the fat framework and moves it to the {project root}/iOS_SDK/OneSignalSDK/Framework folder
-# 
-# INPUT ENVIRONMENTAL VARIABLES (set in Xcode project settings for our Aggregate targets)
-#     $ONESIGNAL_DESTINATION_PATH: The path where the final fat framework should be copied to
-#            eg: ${SRCROOT}/Framework
-#
-#     $ONESIGNAL_OUTPUT_NAME: The name of the framework produced (ie. {OneSignal}.framework)
-#
-#     $ONESIGNAL_TARGET_NAME: The name of the actual Xcode target that produces the framework
-#
-#     #ONESIGNAL_MACH_O_TYPE: Determines if the project is build as a static or dynamic library
-
+#!/bin/bash
 set -e
-set -o pipefail
 
-# Build x86 based framework to support iOS simulator 
-xcodebuild -configuration "${CONFIGURATION}" -project "${PROJECT_NAME}.xcodeproj" -target ${ONESIGNAL_TARGET_NAME} -sdk "iphonesimulator${SDK_VERSION}" "${ACTION}" ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode RUN_CLANG_STATIC_ANALYZER=NO CLANG_ENABLE_MODULE_DEBUGGING=NO BUILD_DIR="${BUILD_DIR}" BUILD_ROOT="${BUILD_ROOT}" SYMROOT="${SYMROOT}" MACH_O_TYPE=${ONESIGNAL_MACH_O_TYPE} -UseModernBuildSystem=NO
+WORKING_DIR=$(pwd)
+
+# For backwards compatible bitcode we need to build iphonesimulator + iphoneos with 3 versions behind the latest.
+#       However variant=Mac Catalyst needs to be be Xcode 11.0
+/Applications/Xcode10.1.app/Contents/Developer/usr/bin/xcodebuild -configuration "Debug" MACH_O_TYPE="staticlib" -sdk "iphonesimulator" ARCHS="x86_64" -project "OneSignal.xcodeproj" -scheme "OneSignalFramework" SYMROOT="temp/"
+/Applications/Xcode10.1.app/Contents/Developer/usr/bin/xcodebuild -configuration "Debug" MACH_O_TYPE="staticlib" -sdk "iphoneos" ARCHS="armv7 armv7s arm64 arm64e i386"  -project "OneSignal.xcodeproj" -scheme "OneSignalFramework" SYMROOT="temp/"
+xcodebuild -configuration "Debug" ARCHS="x86_64h" -destination 'platform=macOS,variant=Mac Catalyst' MACH_O_TYPE="staticlib" -project "OneSignal.xcodeproj" -scheme "OneSignalFramework" SYMROOT="temp/"
+
+USER=$(id -un)
+DERIVED_DATA_ONESIGNAL_DIR="${WORKING_DIR}/temp"
```

Comment:
Let's make `temp` here a variable and use it in the 3 `xcodebuild` commands above. Name suggestion, `DERIVED_DATA_RELITIVE_DIR`